swap/src/protocol/bob/swap.rs

```diff
@@ -41,9 +40,8 @@ pub async fn run_until(
         )
         .await?;
 
-        let db_state = current_state.clone().into();
         swap.db
-            .insert_latest_state(swap.id, Swap::Bob(db_state))
+            .insert_latest_state(swap.id, &current_state)
```

Comment:
If we are saving `current_state` directly now, doesn't this mean we would be able to delete `database::Swap`?